src/setup/node.rs

```diff
@@ -0,0 +1,167 @@
+//! High level APIs and types for node setup and teardown.
+
+use std::{
+    fs, io,
+    path::{Path, PathBuf},
+    process::{Child, Command, Stdio},
+};
+
+use anyhow::Result;
+use fs_extra::dir;
+
+use super::constants::{ALGORAND_SETUP_DIR, NODE_DIR, PRIVATE_NETWORK_DIR};
+use crate::setup::{
+    config::{NodeConfig, NodeMetaData},
+    get_algorand_work_path,
+};
+
+pub enum ChildExitCode {
+    Success,
+    ErrorCode(Option<i32>),
+}
+
+pub struct NodeBuilder {
+    /// Node's startup configuration.
+    conf: NodeConfig,
+    /// Node's process metadata read from Ziggurat configuration files.
+    meta: NodeMetaData,
+}
+
+impl NodeBuilder {
+    /// Creates a new [NodeBuilder].
+    pub fn new() -> anyhow::Result<Self> {
+        let setup_path = get_algorand_work_path()?.join(ALGORAND_SETUP_DIR);
+
+        let conf = NodeConfig::default();
+        let meta = NodeMetaData::new(setup_path)?;
+
+        Ok(Self { conf, meta })
+    }
+
+    /// Creates a [Node] according to configuration and starts its process.
+    pub async fn start(&mut self, target: &Path) -> Result<Node> {
+        if !target.exists() {
+            fs::create_dir_all(&target)?;
+        }
+        let setup_path = get_algorand_work_path()?.join(ALGORAND_SETUP_DIR);
+
+        // Currently we can start only the first node
+        let source = get_node_path(0)?;
+
+        let mut copy_options = dir::CopyOptions::new();
+        copy_options.content_only = true;
+        copy_options.overwrite = true;
+        dir::copy(&source, &target, &copy_options)?;
+
+        // TODO(Rqnsom) configure the node
+
+        self.meta.start_args.push("-d".into());
+        self.meta.start_args.push(target.into());
+
+        if self.conf.log_to_stdout {
+            self.meta.start_args.push("-o".into());
+        }
+
+        let node = self.start_node();
+
+        // TODO(Rqnsom) wait for the connection to confirm
+
+        self.meta = NodeMetaData::new(setup_path)?; // Reset args
```

Comment:
why? needs a comment too

Reply:
Will be removed after doing `bulder->spawn`, `node->start`.